identity/handler.go

```diff
@@ -205,6 +210,30 @@ type adminCreateIdentity struct {
 	Body AdminCreateIdentityBody
 }
 
+// CredentialsConfig is the struct that is being used as part of the identity credentials.
+type CredentialsConfig struct {
+	// HashedPassword is a hash-representation of the password.
+	HashedPassword string `json:"hashed_password"`
+}
+
+// User's password
+//
+// The password of the user. It represents a cleartext or hashed user password.
+//
+// swagger:model passwordCredential
+type PasswordCredential struct {
+	// Value represents the value of the password in clear text or hashed
+	//
+	// required: true
+	Value string `json:"value"`
+
+	// This field must be true if the value represents a hashed password. Otherwise the value is assumed to be a
+	// cleartext password and it will be hashed before storing.
+	//
+	// required: true
+	IsHashed bool `json:"is_hashed"`
```

Comment:
This looks great! I have a short idea on the API design:

```
{
  traits: { ... }
  credentials: {
    "password": {
      "hash": "...", // OR
      "cleartext": "..."
    }
  }
}
```

That would allow us to also add more import types in the future, such as importing OIDC credentials:

```
{
  traits: { ... }
  credentials: {
    "oidc": {
      "providers:" [ ... ]
    }
  }
}
```

WDYT?